code/controllers/subsystem/jobs.dm

```diff
@@ -726,3 +726,135 @@ SUBSYSTEM_DEF(jobs)
 		new_id_change_records["[id_change_counter]"] = thisrecord
 		id_change_counter++
 	id_change_records = new_id_change_records
+
+// This proc will update all players EXP at once. It will calculate amount of time to add dynamically based on the SS fire time.
+/datum/controller/subsystem/proc/batch_update_player_exp(announce = FALSE)
```

Comment:
Bear in mind there's no way for announce to be = TRUE in my code.
All the stuff that uses announce is there for debugging purposes (IE: so you can proccall it and verify its assigning your exp correctly). You probably already know this, but I mention it in case it helps you with testing/debugging.